src/dom.ts

```diff
@@ -158,3 +179,52 @@ export function isVisible (element: Element) {
     /* } */
     /* return true */
 }
+
+/** Get all elements that match the given selector
+ *
+ * @param selector   `the CSS selector to choose elements with
+ * @param filter      filter to use to further chose items, or null for all
+ */
+export function getElemsBySelector(selector: string, filter: ElementFilter) {
+    let elems = Array.from(document.querySelectorAll(selector))
+
+    return filter ? elems.filter(filter) : elems
+}
+
+/** Get the nth input element on a page
+ *
+ * @param nth         the element index, can be negative to start at the end
+ * @param filter      filter to use to further chose items, or null for all
+ */
+export function getNthElement(selectors: string, nth: number,
+    filter: ElementFilter) {
+
+    let inputs = getElemsBySelector(selectors, filter)
+
+    if (inputs.length) {
+        let index = Number(nth).clamp(-inputs.length, inputs.length - 1)
+            .mod(inputs.length)
+
+        return <HTMLElement>inputs[index]
+    }
+
+    return null
+}
+
+/** Get the largest element out of an array of Elements
+ */
+export function getLargestElement(elems: Element[]) {
```

Comment:
I think a neater way to do this would be more like completions.ts:321 - use sort and then pick the first element.

I'm happy to be persuaded otherwise : )